packages/edit-site/src/components/global-styles/font-library-modal/upload-fonts.js

```diff
@@ -45,29 +45,50 @@ function UploadFonts() {
 	 * @param {Array} files The files to be filtered
 	 * @return {void}
 	 */
-	const handleFilesUpload = ( files ) => {
+	const handleFilesUpload = async ( files ) => {
 		setNotice( null );
 		setIsUploading( true );
 		const uniqueFilenames = new Set();
 		const selectedFiles = [ ...files ];
-		const allowedFiles = selectedFiles.filter( ( file ) => {
+		let hasInvalidFiles = false;
+
+		// Use map to create a promise for each file check, then filter with Promise.all.
+		const checkFilesPromises = selectedFiles.map( async ( file ) => {
+			const isFont = await isFontFile( file );
+			if ( ! isFont ) {
+				hasInvalidFiles = true;
+				return null; // Return null for invalid files.
+			}
+			// Check for duplicates
 			if ( uniqueFilenames.has( file.name ) ) {
-				return false; // Discard duplicates
+				return null; // Return null for duplicates.
 			}
-			// Eliminates files that are not allowed
+			// Check if the file extension is allowed.
 			const fileExtension = file.name.split( '.' ).pop().toLowerCase();
 			if ( ALLOWED_FILE_EXTENSIONS.includes( fileExtension ) ) {
 				uniqueFilenames.add( file.name );
-				return true; // Keep file if the extension is allowed
+				return file; // Return the file if it passes all checks.
 			}
-			return false; // Discard file extension not allowed
+			return null; // Return null for disallowed file extensions.
 		} );
+
+		// Filter out the nulls after all promises have resolved.
+		const allowedFiles = ( await Promise.all( checkFilesPromises ) ).filter(
+			( file ) => null !== file
+		);
+
 		if ( allowedFiles.length > 0 ) {
 			loadFiles( allowedFiles );
 		} else {
+			const message = hasInvalidFiles
+				? __(
+						'One or more of the selected files are not valid fonts. Please try again.'
+				  )
+				: __( 'No valid fonts found to install.' );
+
```

Comment:
@getdave I'm adding a separate error message to let the user know that they're trying to upload invalid files. Do you think this is reasonably justifiable if we were to include the bugfix in 6.5?

Reply:
I suspect that regarding the 6.5 string freeze, it's somewhat the same to include a new string rather than modify an existing one.

Given both messages say basically the same, I'd prefer to simplify the code by removing this ternary and keeping only one string. That would allow us to remove these lines, too:

https://github.com/WordPress/gutenberg/blob/39228ab9464be8b871dc7b0c3637bf05ffa5369b/packages/edit-site/src/components/global-styles/font-library-modal/upload-fonts.js#L59

https://github.com/WordPress/gutenberg/blob/39228ab9464be8b871dc7b0c3637bf05ffa5369b/packages/edit-site/src/components/global-styles/font-library-modal/upload-fonts.js#L53

Reply:
It might be possible to adapt some existing strings in the WP 6.5 branch:

`__( 'There were some errors installing fonts. %s' )` 
`__( 'No fonts found to install.' )`
`__( 'Sorry, you are not allowed to upload this file type.' )`

Reply:
For 6.5 let's try and reuse strings we already have. We can optimise to a better string later (perhaps create a followup Issue?).